android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.kt

```diff
@@ -263,57 +265,97 @@ class ScreenStackFragment :
         enter: Boolean,
         nextAnim: Int,
     ): Animator? {
-        if (!screen.usesFormSheetPresentation()) {
-            // Use animation defined while defining transaction in screen stack
-            return null
-        }
+        var animatorSet = AnimatorSet()
 
-        val animatorSet = AnimatorSet()
-        val dimmingDelegate = requireDimmingDelegate()
-
-        if (enter) {
-            val alphaAnimator =
-                ValueAnimator.ofFloat(0f, dimmingDelegate.maxAlpha).apply {
-                    addUpdateListener { anim ->
-                        val animatedValue = anim.animatedValue as? Float
-                        animatedValue?.let { dimmingDelegate.dimmingView.alpha = it }
-                    }
+        if (!screen.usesFormSheetPresentation()) {
+            when (nextAnim) {
+                R.animator.rns_default_exit_out,
+                R.animator.rns_default_exit_in,
+                R.animator.rns_default_enter_out,
+                R.animator.rns_default_enter_in,
+                R.animator.rns_no_animation_20,
+                R.animator.rns_fade_out,
+                R.animator.rns_fade_in,
+                R.animator.rns_slide_out_to_left,
+                R.animator.rns_slide_in_from_left,
+                R.animator.rns_slide_out_to_right,
+                R.animator.rns_slide_in_from_right,
```

Comment:
I'm also not convinced that this is the place to customise this. What we need is cohesive approach to setting up animations. Right now we set them both here & in `ScreenStack.onUpdate`. This is kinda bad & we need to come up with some clear abstraction here. I don't have clear idea in mind yet. Will come back here once I think this over.